src/LanguageServer/Microsoft.CodeAnalysis.LanguageServer/HostWorkspace/OpenProjectsHandler.cs

```diff
@@ -31,7 +31,7 @@ public OpenProjectHandler(LanguageServerProjectSystem projectSystem)
 
     Task INotificationHandler<NotificationParams, RequestContext>.HandleNotificationAsync(NotificationParams request, RequestContext requestContext, CancellationToken cancellationToken)
     {
-        return _projectSystem.OpenProjectsAsync(request.Projects.SelectAsArray(p => p.LocalPath));
+        return _projectSystem.OpenProjectsAsync(request.Projects.SelectAsArray(p => (ProjectPath: p.LocalPath, ProjectGuid: (string?)null)));
```

Comment:
Not a huge deal, but this seems like this should be kept inside the LanguageServerProjectSystem implementation; not sure why ProjectToLoad would be leaking here as a direct type.

Reply:
ProjectToLoad is not being used here. These are tuples. But I did condense the overloads now so that you always have to pass something for ProjectGuid. The Guid-less overload can be brought back if you prefer.

Reply:
I think the need for this overload might have been due to some intermediate state, it's not necessary now. I'm reverting the change to this line.